contracts/src/libraries/xChain/T1XChainReader.sol

```diff
@@ -191,53 +189,35 @@ contract T1XChainReader is OwnableUpgradeable, ReentrancyGuardUpgradeable {
 
     /**
      * @notice Handles incoming messages from other chains
-     * @param _originDomain The origin domain
-     * @param _sender The sender address on the target chain
-     * @param _requestId The ID assigned to the request when it was dispatched
-     * @param _data The data to relay to the callback contract
+     * @param requestId The ID assigned to the request when it was dispatched
+     * @param batchIndex The batch index of the read request
+     * @param newRoot The root of the proof of read merkle tree
      */
-    function handle(
-        uint32 _originDomain,
-        bytes32 _sender,
-        bytes32 _requestId,
-        bytes memory _data
-    )
-        external
-        payable
-        onlyProver
-    {
-        _handleReadResponse(_originDomain, _sender, _requestId, _data);
+    function handle(bytes32 requestId, uint256 batchIndex, bytes32 newRoot) external payable onlyProver {
+        _handleReadResponse(requestId, batchIndex, newRoot);
```

Comment:
IMO it would be better to store `nextBatchIndex` on-chain and increment it similar to the `T1Chain` contract.